packages/react/src/utilities/ThemeProvider/makeStyles.ts

```diff
@@ -61,15 +62,38 @@ export function makeStyles<TStyleSet extends { [key in keyof TStyleSet]: IStyle
   // eslint-disable-next-line deprecation/deprecation
   return (options: UseStylesOptions = {}) => {
     let { theme } = options;
-    const win = useWindow();
+    let winId: string | undefined;
+    // Retain a dictionary of window ids we're tracking
+    const allWindows = new Set<string>();
+    const win = useWindow() as (Window & { __id__: string }) | undefined;
+    if (win) {
+      win.__id__ = win.__id__ || getId();
+      winId = win.__id__;
+      if (!allWindows.has(winId)) {
+        allWindows.add(winId);
+        win.addEventListener('unload', () => cleanupMapEntries(winId));
+      }
+      // cleanupMapEntries will
+      // 1. remove all the graph branches for the window,
+      // 2. remove the event listener,
+      // 3. delete the allWindows entry.
+      const cleanupMapEntries = (id: string | undefined) => {
+        if (id) {
+          graph.delete(winId);
+          win.removeEventListener('unload', () => cleanupMapEntries(id));
```

Comment:
This isn't going to work. You're passing in a new function which isn't the same function as the one you hooked up. The `removeEventListener` requires the same function you passed into `addEventListener`.

Instead, above the return, create this function:

```js
const cleanupMapEntries(ev: HTMLUnloadEvent) {
  const win = ev.target as WindowWithId;
  const winId = win.__id__;

  // now clean up using the above.
}
```

This means that add/removing the event listener doesn't require a lambda:

```js
win.addEventListener('unload', cleanupMapEntries)
win.removeEventListener('unload', cleanupMapEntries)
```

Reply:
Yeah, i was wondering about passing the ID in. It didn't seem right.